packages/flutter_tools/lib/src/project.dart

```diff
@@ -167,33 +151,52 @@ class IosProject {
     final File projectFile = directory.childDirectory('Runner.xcodeproj').childFile('project.pbxproj');
     return _firstMatchInFile(projectFile, _productBundleIdPattern).then((Match match) => match?.group(1));
   }
-}
-
-/// Represents the contents of the .ios/ folder of a Flutter module
-/// project.
-class IosModuleProject {
-  IosModuleProject(this.directory);
 
-  final Directory directory;
-
-  Future<void> ensureReadyForPlatformSpecificTooling() async {
-    if (_shouldRegenerate()) {
+  Future<void> ensureReadyForPlatformSpecificTooling(FlutterProject parent) async {
+    if (isModule && _shouldRegenerateFromTemplate()) {
       final Template template = new Template.fromName(fs.path.join('module', 'ios'));
       template.render(directory, <String, dynamic>{}, printStatusWhenWriting: false);
     }
+    if (!directory.existsSync())
+      return;
+    if (Cache.instance.fileOlderThanToolsStamp(generatedXcodePropertiesFile)) {
+      await xcode.updateGeneratedXcodeProperties(
+        project: parent,
+        buildInfo: BuildInfo.debug,
+        targetOverride: bundle.defaultMainPath,
+        previewDart2: true,
+      );
+    }
   }
 
-  bool _shouldRegenerate() {
+  bool _shouldRegenerateFromTemplate() {
     return Cache.instance.fileOlderThanToolsStamp(directory.childFile('podhelper.rb'));
   }
+
+  File get generatedXcodePropertiesFile => directory.childDirectory('Flutter').childFile('Generated.xcconfig');
+
+  Directory get pluginRegistrantHost {
+    return isModule
+        ? directory.childDirectory('Flutter').childDirectory('FlutterPluginRegistrant')
+        : directory.childDirectory('Runner');
+  }
 }
 
-/// Represents the contents of the android/ folder of a Flutter project.
+/// Represents the Android sub-project of a Flutter project.
+///
+/// Instances will reflect the contents of the `android/` sub-folder of
```

Comment:
What will happen after running `flutter materialize`? I thought that was the whole reason to have two types from the beginning.
I guess the different subdirectories can be made to point into `android/blah/` and `.android/foo` respectively, but `AndroidProject.directory` becomes ambiguous.
But then again, perhaps we never use it on the outside (in that case it should be made private).

Reply:
Yes, I'll work towards encapsulating it.